faker/providers/file/__init__.py

```diff
@@ -279,25 +292,38 @@ def file_path(
         or an empty sequence (the path will have no extension). Default
         behaviour is the same as |file_name|
 
+        if ``file_system`` is set (default="linux"), the generated path uses
+        specified file system path standard, the list of valid file systems include:
+        ``'windows'``, ``'linux'``.
+
         :sample: size=10
         :sample: depth=3
         :sample: depth=5, category='video'
         :sample: depth=5, category='video', extension='abcdef'
         :sample: extension=[]
         :sample: extension=''
         :sample: extension=["a", "bc", "def"]
+        :sample: depth=5, category='video', extension='abcdef', file_system='windows'
         """
+
         if extension is not None and not isinstance(extension, str):
             if len(extension):
                 extension = self.random_element(extension)
             else:
                 extension = ""
 
-        file: str = self.file_name(category, extension)
-        path: str = f"/{file}"
+        fs_rule = self.file_systems_path_rules.get(file_system_rule, None)
+        if not fs_rule:
+            raise TypeError("Specified file system is invalid.")
+
+        root: str = fs_rule.get("root")
+        seperator: str = fs_rule.get("seperator")
```

Comment:
Since we're defining the values of the dictionary ourselves, I dont think we need to worry about non-existent keys.

```suggestion
        root = fs_rule["root"]
        seperator = fs_rule["separator"]
```